production_entry_app/production_entry_app/overrides/test_joint_production.py

```
@@ -0,0 +1,877 @@
+			)
+		]
+		incoming_value = sum(value for value in stock_value_differences if value > 0)
+		outgoing_value = -sum(value for value in stock_value_differences if value < 0)
+		currency_precision = int(frappe.db.get_single_value("System Settings", "currency_precision") or 2)
+		self.assertAlmostEqual(
+			incoming_value - outgoing_value,
+			100,
+			delta=(10**-currency_precision) * len(stock_value_differences),
+		)
+
+	def test_api_e2e_shift_start_joint_submit_and_cancel_reverses_everything(self) -> None:
```

Ruff (0.16.1)

[warning] Too many statements (61 > 50)

(PLR0915)